dbms/src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -355,8 +355,12 @@ struct StorageDistributedDirectoryMonitor::Batch
             /// we must try to re-send exactly the same batches.
             /// So we save contents of the current batch into the current_batch_file_path file
             /// and truncate it afterwards if all went well.
-            WriteBufferFromFile out{parent.current_batch_file_path};
-            writeText(out);
+            {
+                WriteBufferFromFile out{parent.current_batch_file_path + ".tmp"};
```

Comment:
What will happen if the file already exists? Need a comment that it will truncate the existing file.
Also need a comment about what will hapen with obsolete .tmp files (nothing and user should clean manually).

Reply:
> Also need a comment about what will hapen with obsolete .tmp files (nothing and user should clean manually).

Should user care about these `.tmp` files, if they will be simply overwritten?

Reply:
> Should user care about these .tmp files, if they will be simply overwritten?

No. But as far as I remember, we use auto-increment, so the next file will have different name.

Reply:
Hm, [https://github.com/ClickHouse/ClickHouse/blob/5aceca9e178ec07638a07c4b28f42b26ab005268/dbms/src/Storages/Distributed/DirectoryMonitor.cpp#L63-L65](don't see) the auto-increment bit.